internal/pass/sign/sign.go

```diff
@@ -0,0 +1,108 @@
+package sign
+
+import (
+	"crypto"
+	"encoding/asn1"
+	"fmt"
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/golang-jwt/jwt/v5"
+)
+
+type Message struct {
+	ConnectionID   string
+	ExpiresAt      time.Time
+	ConnectionType ConnectionType
+}
+
+// SignAndEncode information in the message. The result
+// is second and third part of jwt token. Since the first
+// part is constant it is omitted.
+func (s Service) SignAndEncode(m Message) (string, error) {
+	token := jwt.NewWithClaims(s.signingMethod, jwt.MapClaims{
+		"exp":  m.ExpiresAt.Unix(),
+		"aud":  []string{string(m.ConnectionType)},
+		"c_id": m.ConnectionID,
+	})
+
+	// no key is needed, as we use custom signing method.
+	signed, err := token.SignedString(nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to sign jwt: %w", err)
+	}
+
+	if !strings.HasPrefix(signed, jwtHeader) {
+		return "", fmt.Errorf("unpexpected signed string format")
+	}
+
+	return strings.TrimPrefix(signed, jwtHeader), nil
+}
+
+type kmsSigningMethod struct {
+	client *kms.KMS
+	keyID  string
+	hash   crypto.Hash
+}
+
+// Verify implements jwt.SigningMethod#Method. Because we
+// provide key to jwt library, this is never called.
+func (s kmsSigningMethod) Verify(signingString string, sig []byte, key interface{}) error {
+	panic("should never be called")
+}
+
+// Sign implements jwt.SigningMethod#Sign method.
+func (s kmsSigningMethod) Sign(signingString string, key interface{}) ([]byte, error) {
+	messageType := "DIGEST"
+
+	hasher := s.hash.New()
+	if _, err := hasher.Write([]byte(signingString)); err != nil {
+		return nil, fmt.Errorf("failed to hash input")
+	}
+	hashedSigningString := hasher.Sum(nil)
+
+	resp, err := s.client.Sign(&kms.SignInput{
+		KeyId:            &s.keyID,
+		Message:          hashedSigningString,
+		MessageType:      &messageType,
+		SigningAlgorithm: aws.String(awsSigningAlgorithm),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign the message: %w", err)
+	}
+	keySizeInBytes := 258 / 8
```

Comment:
can you add comment why those numbers? i think it should be 256/8 , Those are taken fromhttps://github.com/matelang/jwt-go-aws-kms/blob/main/jwtkms/kms_signing_method.go#L49?

Reply:
done. And you are right about the number.